src/odemis/driver/test/orsay_test.py

```diff
@@ -0,0 +1,652 @@
+        self.assertIsInstance(self.psus.state.value, HwError)
+        self.assertIn("ValvePneumaticSuspension is in error", str(self.psus.state.value))
+        self.psus._valve.Target = -1
+        sleep(1)
+        self.assertIsInstance(self.psus.state.value, HwError)
+        self.assertIn("ValvePneumaticSuspension could not be contacted", str(self.psus.state.value))
+        self.psus._valve.Target = 1
+        sleep(5)
+        self.assertEqual(self.psus.state.value, model.ST_RUNNING)
+
+    def test_updatePower(self):
+        """
+        Check that the power VA is updated correctly and an exception is raised when the wrong parameter is passed
+        """
+        with self.assertRaises(ValueError):
+            self.psus._updatePower(self.datamodel.HybridPlatform.Cancel)
+
+        self.psus._valve.Target = 1
+        sleep(1)
+        self.assertTrue(self.psus.power.value)
+
+        self.psus._valve.Target = 2
+        sleep(1)
+        self.assertFalse(self.psus.power.value)
+
+    def test_updatePressure(self):
+        """
+        Check that the pressure VA is updated correctly and an exception is raised when the wrong parameter is passed
+        """
+        with self.assertRaises(ValueError):
+            self.psus._updatePressure(self.datamodel.HybridPlatform.Cancel)
+
+        if not TEST_NOHW == "sim":
+            self.skipTest("TEST_NOHW is not set to sim, cannot force data on Actual parameters of Orsay server "
+                          "outside of simulation.")
+        test_value = 1.0
+        self.psus._gauge.Actual = test_value
+        sleep(1)
+        self.assertEqual(self.psus.pressure.value, test_value)
+        self.psus._gauge.Actual = 0.0
+
+
+class TestVacuumChamber(unittest.TestCase):
+    """
+    Tests for the vacuum chamber
+    """
+
+    oserver = None
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Setup the Orsay client
+        """
+        if TEST_NOHW == 1:
+            raise unittest.SkipTest("TEST_NOHW is set. No server to contact.")
+        try:
+            cls.oserver = orsay.OrsayComponent(**CONFIG_ORSAY)
+        except Exception as e:
+            raise HwError(str(e))
+
+        cls.datamodel = cls.oserver.datamodel
+
+        for child in cls.oserver.children.value:
+            if child.name == CONFIG_PRESSURE["name"]:
+                cls.pressure = child
+
+    @classmethod
+    def tearDownClass(cls):
+        """
+        Terminate the Orsay client
+        """
+        cls.oserver.terminate()
+
+    def test_valve(self):
+        """
+        Test for controlling the gate valve of the chamber
+        """
+        self.pressure._gate.IsOpen.Target = 1
+        sleep(1)
+        self.assertTrue(self.pressure.gateOpen.value)
+
+        self.pressure._gate.IsOpen.Target = 2
+        sleep(1)
+        self.assertFalse(self.pressure.gateOpen.value)
+
+        self.pressure.gateOpen.value = True
+        self.assertEqual(self.pressure._gate.IsOpen.Target, 1)
```

Comment:
You should also check that reading self.pressure.gateOpen.value after 1 sec still reports the right value.